include/eve/detail/function/simd/riscv/friends.hpp

```diff
@@ -0,0 +1,120 @@
+  {                                                                                                \
+    if constexpr( std::is_signed_v<T> ) return int_func(lhs, rhs, N::value);                       \
+    else return unsigned_func(lhs, rhs, N::value);                                                 \
+  }                                                                                                \
+  template<plain_scalar_value T, typename N>                                                       \
+  EVE_FORCEINLINE auto eve_name(wide<T, N> lhs, std::convertible_to<T> auto rhs) noexcept          \
+      -> logical<wide<T, N>>                                                                       \
+  requires rvv_abi<abi_t<T, N>> && (!std::is_floating_point_v<T>)                                  \
+  {                                                                                                \
+    if constexpr( std::is_signed_v<T> ) return int_func(lhs, static_cast<T>(rhs), N::value);       \
+    else return unsigned_func(lhs, static_cast<T>(rhs), N::value);                                 \
+  }                                                                                                \
+  template<plain_scalar_value T, typename N>                                                       \
+  EVE_FORCEINLINE auto eve_name(wide<T, N> lhs, wide<T, N> rhs) noexcept -> logical<wide<T, N>>    \
+  requires rvv_abi<abi_t<T, N>> && (std::is_floating_point_v<T>)                                   \
+  {                                                                                                \
+    logical<wide<T, N>> res = fp_func(lhs, rhs, N::value);                                         \
+    return res;                                                                                    \
+  }                                                                                                \
+  template<plain_scalar_value T, typename N>                                                       \
+  EVE_FORCEINLINE auto eve_name(wide<T, N> lhs, std::convertible_to<T> auto rhs) noexcept          \
+      -> logical<wide<T, N>>                                                                       \
+  requires rvv_abi<abi_t<T, N>> && (std::is_floating_point_v<T>)                                   \
+  {                                                                                                \
+    return fp_func(lhs, static_cast<T>(rhs), N::value);                                            \
+  }
+
+RVV_LOGIC(self_greater, __riscv_vmsgt, __riscv_vmsgtu, __riscv_vmfgt)
+RVV_LOGIC(self_less, __riscv_vmslt, __riscv_vmsltu, __riscv_vmflt)
+RVV_LOGIC(self_geq, __riscv_vmsge, __riscv_vmsgeu, __riscv_vmfge)
+RVV_LOGIC(self_leq, __riscv_vmsle, __riscv_vmsleu, __riscv_vmfle)
+RVV_LOGIC(self_eq, __riscv_vmseq, __riscv_vmseq, __riscv_vmfeq)
+RVV_LOGIC(self_neq, __riscv_vmsne, __riscv_vmsne, __riscv_vmfne)
```

Comment:
please no macros like this. We can figure out template to do this if needed.

Reply:
Removed